src/odemis/acq/stream/_sync.py

```diff
@@ -812,25 +823,45 @@ def _estimateRawAcquisitionTime(self):
             res = self._sccd._getDetectorVA("resolution").value
             readout = numpy.prod(res) / ro_rate
 
-            exp = self._sccd._getDetectorVA("exposureTime").value
+            if hasattr(self._sccd, "integrationTime"):
+                exp = self._sccd.integrationTime.value
+                # calculate number of images to integrate
+                hwExpTimeMax = self._sccd.detector.exposureTime.range[1]
+                numImages2integrate = int(numpy.ceil(exp / hwExpTimeMax))
+                readout *= numImages2integrate
+            else:
+                exp = self._sccd._getDetectorVA("exposureTime").value
+
             dur_image = (exp + readout + 0.03) * 1.20
             duration = numpy.prod(self.repetition.value) * dur_image
             # Add the setup time
             duration += self.SETUP_OVERHEAD
 
             return duration
+
         except Exception:
             msg = "Exception while estimating acquisition time of %s"
             logging.exception(msg, self.name.value)
+
             return Stream.estimateAcquisitionTime(self)
 
     def _adjustHardwareSettings(self):
         """
-        Read the SEM and CCD stream settings and adapt the SEM scanner
-        accordingly.
-        return (float): estimated time for a whole CCD image
+        Read the SEM and CCD stream settings and adapt the SEM scanner accordingly.
+        :returns: exp + readout (float): Estimated time for a whole (but not integrated) CCD image.
         """
-        exp = self._sccd._getDetectorVA("exposureTime").value  # s
+        if model.hasVA(self._sccd, "integrationTime"):
+            # calculate exposure time to be set on detector
+            exp = self._sccd.integrationTime.value / self._sccd.integrationCounts.value
```

Comment:
I'm not very convinced by the idea of using `self._sccd.integrationCounts.value`. Primarily, because it's coming from a separate class, which has this VA mostly only for display purpose (so, not so important to have correct), and here we rely on it to have the right acquisition parameters. On the other you could argue that this information to the user should always be correct, and if it's not, it should be fixed, and not do different things in the acquisition.

However, I'd really argue against our own `self._integrationCounts`, and instead, I'd suggest to return from this function "exp_time, integ_count". That second parameter would be then passed to the functions which care about it (so it makes the code more explicit).

Reply:
I am confused. We already have a self._integrationCounts VA here. And we set it default to 1 also if no integrationVA. Can we quickly discuss please, also in regard to the next comments?

Reply:
self._integrationCounts is only used in one function: the caller of `_adjustHardwareSettings()`. However, it's computed by a separate class, and there is a slight chance that it's incorrect in case the hardware actually accepts a different exposure time. So I'd suggest that this function returns the "definitive" integ_counts, and that value would be used by the caller.

(And as a tiny bonus, that allows you to remove the couple of lines needed to initialise properly this VA in `__init__`!)